python/requirements/ml/data-test-requirements.txt

```diff
@@ -12,4 +12,5 @@ google-cloud-core==2.4.1
 google-cloud-bigquery-storage==2.24.0
 google-api-core==1.34.0
 webdataset
+pyiceberg>=0.6.0
```

Comment:
can we pin to a specific version, e.g. `pyiceberg==0.6.0`? we prefer to pin to a specific version to keep tests consistent, and not be dependent on potential upstream changes that may break our CI.